kolibri/tasks/api.py

```diff
@@ -101,16 +101,19 @@ def startlocalexport(self, request):
     @list_route(methods=['post'])
     def cleartask(self, request):
         '''
-        Temporary hack to clear all tasks. Should actually take a single task ID.
+        Clears a task with its task id given in the task_id parameter.
         '''
-        import subprocess
-        import os
-        print("CLEAR TASKS HACK")
-        subprocess.check_output([
-            "sqlite3",
-            os.path.expanduser("~/.kolibri/ormq.sqlite3"),
-            "delete from django_q_task; delete from django_q_ormq;"
-        ])
+
+        if 'task_id' not in request.data:
+            raise serializers.ValidationError("The 'task_id' field is required.")
+
+        task_id = request.data['task_id']
+
+        # we need to decrypt tasks first to get their real task_id. Hence why this python-side task_id retrieval and deletion.
+        [taskitem.delete() for taskitem in OrmQ.objects.all() if taskitem.task()["id"] == task_id]
+
+        Task.objects.filter(pk=task_id).delete()
```

Comment:
This removes the results -- `Task` is for the task results (where the progress data is stored), `OrmQ` is for the queues.